### PR TITLE
adding serializers

### PR DESCRIPTION
**Pull request by marinaRomanchuk**

### CODE REVIEW

File: comments/serializers.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+from rest_framework import serializers
+
+from .models import Comment
+
+
+class CommentSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Comment
+        fields = ["text", "post", "user", "timestamp"]
```
Comment (revimi) [inline comments/serializers.py:9]: 1. As we are going to have a list of comments shown below the post, we don't need to serialize the `post` attribute for each entry. We can rather have `post_id` field
2. For the `user` field: [Nested relationships in DRF](https://www.django-rest-framework.org/api-guide/relations/#nested-relationships)

Comment (revimi) [inline comments/serializers.py]: 1. As we are going to have a list of comments shown below the post, we don't need to serialize the `post` attribute for each entry. We can rather have `post_id` field
2. For the `user` field: [Nested relationships in DRF](https://www.django-rest-framework.org/api-guide/relations/#nested-relationships)

File: comments/serializers.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+from rest_framework import serializers
+
+from .models import Comment
```
Comment (revimi) [inline comments/serializers.py]: Personally I prefer absolute imports, but you may leave it as is

File: likes/serializers.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
+from rest_framework import serializers
+
+from .models import Dislike, Like
+
+
+class LikeSerializer(serializers.ModelSerializer):
```
Comment (revimi) [inline likes/serializers.py]: Not sure we need serializers for likes/dislikes as we are going to show just count of them

File: posts/serializers.py [hunk 4]
```diff
@@ -0,0 +1,9 @@
+from rest_framework import serializers
+
+from .models import Post
+
+
+class PostSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Post
+        fields = ["photo", "description", "timestamp", "user"]
```
Comment (revimi) [inline posts/serializers.py:12]: For the `user` field: [Nested relationships in DRF](https://www.django-rest-framework.org/api-guide/relations/#nested-relationships)

File: users/serializers.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+            "username",
+            "first_name",
+            "last_name",
+            "description",
+            "date_of_birth",
+            "profile_photo",
+        ]
+
+
+class FollowersSerializer(serializers.ModelSerializer):
```
Comment (revimi) [inline users/serializers.py]: I don't think we are going to use it. We will be showing a list of someone's followers and use `UserSerializer` for that

File: users/serializers.py [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+class UserSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = User
+        fields = [
+            "username",
+            "first_name",
+            "last_name",
+            "description",
+            "date_of_birth",
```
Comment (revimi) [inline users/serializers.py]: Probably users don't want `date_of_birth` to be publicly accessable. So let's have at least two serializers for the User model: 
1. Serializer containing only public user data
2. Serializer containing full user data

They can subclass one another

**Review by revimi — CHANGES_REQUESTED** [review]

File: likes/serializers.py (hunk 3 above)
Comment (marinaRomanchuk) [reply likes/serializers.py]: Remowed

File: posts/serializers.py (hunk 4 above)
Comment (marinaRomanchuk) [reply posts/serializers.py:12]: Done

File: users/serializers.py (hunk 5 above)
Comment (marinaRomanchuk) [reply users/serializers.py]: Remowed

File: users/serializers.py (hunk 6 above)
Comment (marinaRomanchuk) [reply users/serializers.py]: Done

File: comments/serializers.py (hunk 1 above)
Comment (marinaRomanchuk) [reply comments/serializers.py]: Done

File: comments/serializers.py (hunk 2 above)
Comment (marinaRomanchuk) [reply comments/serializers.py]: Done

File: users/serializers.py [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+            "profile_photo",
+        ]
```
Comment (revimi) [inline users/serializers.py]: https://www.django-rest-framework.org/api-guide/serializers/#serializer-inheritance
```suggestion
class UserPublicSerializer(serializers.ModelSerializer):
    class Meta:
        model = User
        fields = [
            "username",
            "first_name",
            "last_name",
            "description",
            "profile_photo",
        ]


class UserPrivateSerializer(UserPublicSerializer):
    class Meta(UserPublicSerializer.Meta):
        fields = [
            "username",
            "first_name",
            "last_name",
            "description",
            "date_of_birth",
            "profile_photo",
        ]
```

**Review by revimi — APPROVED** [review]

File: users/serializers.py (hunk 7 above)
Comment (marinaRomanchuk) [reply users/serializers.py]: Done

**Pull request merged**